apps/patient/src/components/order-summary/OrderSummary.tsx

```diff
@@ -0,0 +1,191 @@
+import { Box, Button, Heading, HStack, Text, VStack } from '@chakra-ui/react';
+import dayjs from 'dayjs';
+import { groupBy } from 'lodash';
+import { getLatestReadyTime } from '../../utils/fulfillmentsHelpers';
+import { Card } from '../Card';
+import { PrescriptionFulfillmentState } from '../../__generated__/graphql';
+
+export interface ExceptionData {
+  message?: string;
+  exceptionType: 'OOS' | 'BACKORDERED' | 'PA_REQUIRED';
+}
+
+export interface FulfillmentData {
+  rxName: string;
+  exceptions: ExceptionData[];
+  pharmacyEstimatedReadyAt?: Date | undefined;
+  state: PrescriptionFulfillmentState;
+}
+
+function exceptionCmp(e1: ExceptionData, e2: ExceptionData) {
+  if (e1.exceptionType === 'PA_REQUIRED') return 1;
+  if (e2.exceptionType === 'PA_REQUIRED') return 2;
+  return e1.exceptionType.localeCompare(e2.exceptionType);
+}
+
+function groupFulfillments(fulfillments: FulfillmentData[]) {
+  const derivedState = fulfillments.map((f) => ({
+    ...f,
+    derivedState:
+      f.exceptions.length === 0
+        ? f.state
+        : f.pharmacyEstimatedReadyAt != null
+        ? ('EXCEPTION_WITH_READY_TIME' as const)
+        : ('EXCEPTION_NO_READY_TIME' as const)
+  }));
+
+  const groupedByDerivedState = groupBy(derivedState, 'derivedState') as {
+    [state in (typeof derivedState)[number]['derivedState']]?: typeof derivedState;
```

Comment:
that's some typescript right there

Reply:
This actually looks super sus. Will revisit later